dciv_bot/module/misc.py

```diff
@@ -579,6 +580,85 @@ async def invite(self, ctx):
         invite = await ctx.channel.create_invite(max_age=0, unique=False)
         await ctx.send(invite.url)
 
+    @commands.command(name='roll')
+    @commands.cooldown(1, config.BOT_COMMAND_COOLDOWN, commands.BucketType.user)
+    async def roll(self, ctx, *args):
+        """Roll some dice
+
+            **Supported Notation**
+            - Dice rolls take the form NdX, where N is the number of dice to roll, and X are the faces of the dice. For example, 1d6 is one six-sided die.
+            - A dice roll can be followed by an Ln or Hn, where it will discard the lowest n rolls or highest n rolls, respectively. So 2d20L1 means to roll two d20s and discard the lower. I.E advantage.
+            - A dice roll can be part of a mathematical expression, such as 1d4 +5.
+
+            **Example:**
+              `-roll 1d6` will roll a d6
+              `-roll (2d20L1) + 1d4 + 5` will roll 2d20s, discard the lower one, and add 1d4 and 5 to the result
+
+            *Full notation can be found here: https://xdice.readthedocs.io/en/latest/dice_notation.html*
+            """
+
+        #Todo:
+        # []: Make the special message system more robust and logical 
+
+
+        # Putting all the arguments in one string
+        # https://stackoverflow.com/questions/55665255/how-do-i-put-multiple-words-in-one-argument-discord-py
+        arg = " ".join(args[:])
+
+        try:
+            # Attempt to parse the user command and roll the dice
+            argument = xdice.Pattern(arg)
+            argument.compile()
+            roll = argument.roll()
+        except (SyntaxError, TypeError, ValueError):
```

Comment:
```suggestion
        except (TypeError, ValueError):
```

I'm not sure whether `SyntaxError` can be thrown in that `try` block.

Reply:
If the user input can't be parsed by the library, it throws a syntax error. I didn't see a workaround aside from the try-except block

Reply:
Ah okay, sorry.